tests/testthat/test-save-separate-chunks.R

```diff
@@ -0,0 +1,231 @@
+                         axis.ticks = element_blank(), 
+                         axis.title.x = element_blank(), 
+                         axis.title.y = element_blank()))
+
+USpolygons <- map_data("state")
+USpolygons$subregion <- NULL
+USpolygons <- subset(USpolygons, region != "district of columbia")
+
+# add state flu
+map_flu <- ldply(unique(state_flu$WEEKEND), function(we) {
+  df <- subset(state_flu, WEEKEND == we)
+  merge(USpolygons, df, by.x = "region", by.y = "state")
+})
+
+p <- ggplot() + 
+  make_text(map_flu, -100, 50, "WEEKEND", "CDC FluView in Lower 48 States ending %s") + 
+  scale_fill_gradient2(low = "white", high = "red", breaks = 0:10, guide = "none") + 
+  theme_opts + 
+  theme_animint(width = 750, height= 500)
+
+test_that("save separate chunks for geom_polygon", {
+  state.map <- p + 
+    geom_polygon(data = map_flu, aes(x = long, y = lat, group = group, fill = level, 
+                                     showSelected = WEEKEND), 
+                 colour = "black", size = 1)
+  viz <- list(levelHeatmap = level.heatmap, stateMap = state.map, title = "FluView")
+  out.dir <- file.path(getwd(), "FluView")
+  animint2dir(viz, out.dir = out.dir, open.browser = FALSE)
+  common.chunk <- list.files(path = out.dir, pattern = "geom.+polygon.+chunk_common.tsv", 
+                             full.names = TRUE)
+  varied.chunks <- list.files(path = out.dir, pattern = "geom.+polygon.+chunk[0-9]+.tsv", 
+                              full.names = TRUE)
+  # number of chunks
+  expect_equal(length(common.chunk), 1L)
+  no.chunks <- length(varied.chunks)
+  expect_equal(no.chunks, length(unique(map_flu$WEEKEND)))
+  # test common.chunk
+  common.data <- read.csv(common.chunk, sep = "\t")
+  expect_equal(nrow(common.data), nrow(USpolygons))
+  expect_true(all(c("x", "y", "group") %in% names(common.data)))
+  # randomly choose an varied.chunk to test
+  idx <- sample(no.chunks, 1)
+  varied.data <- read.csv(varied.chunks[idx], sep = "\t")
+  expect_equal(nrow(varied.data), length(unique(USpolygons$group)))
+  expect_true(all(c("fill", "group") %in% names(varied.data)))
+  
+  unlink(out.dir, recursive = TRUE)
+})
+
+
+### test case 2
+USdots <- ddply(USpolygons, .(region), summarise, mean.lat = mean(lat), 
+                mean.long = mean(long))
+# add state flu
+map_flu <- ldply(unique(state_flu$WEEKEND), function(we) {
+  df <- subset(state_flu, WEEKEND == we)
+  merge(USdots, df, by.x = "region", by.y = "state")
+})
+
+test_that("save separate chunks for geom_point without group", {
+  # the compiler will not break a geom into chunks if any of the resulting 
+  # chunk tsv files is estimated to be less than 4KB.
+  state.map <- p + 
+    geom_point(data = map_flu, aes(x = mean.long, y = mean.lat, fill = level, 
+                                   showSelected = WEEKEND), color = "black", size = 10)
+  viz <- list(levelHeatmap = level.heatmap, stateMap = state.map, title = "FluView")
```

Comment:
it would make for a simpler test if yo combined this test "geom_point without group" with the above test "geom_polygon" -- all you would have to do is add the geom_point to the above plot. And then you could test both the polygon tsv files and the point tsv files.